### PR TITLE
Add vaadin-archetype-application for Flow

### LINKED ISSUE #148: Add vaadin-archetype-application for Vaadin platform
Based on https://github.com/viritin/vaadin-flow-archetypes

### PR DESCRIPTION
**Pull request by pleku**
Fixes #148

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1,40 +1,46 @@
...
     <groupId>com.vaadin</groupId>
-    <artifactId>vaadin-parent</artifactId>
-    <version>1.1.0</version>
-  </parent>
-  <groupId>com.vaadin</groupId>
-  <artifactId>vaadin-archetypes</artifactId>
-  <version>8.1-SNAPSHOT</version>
-  <packaging>pom</packaging>
+    <artifactId>vaadin-archetypes</artifactId>
+    <version>13.0.1</version>
```
Comment (mstahv) [inline pom.xml]: 13.0.2 was just out
```suggestion
    <version>13.0.2</version>
```

File: vaadin-archetype-application/pom.xml [hunk 2]
```diff
@@ -1,100 +1,58 @@
...
-  </licenses>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>com.vaadin</groupId>
+        <artifactId>vaadin-archetypes</artifactId>
+        <version>13.0.1</version>
```
Comment (mstahv) [inline vaadin-archetype-application/pom.xml]:
```suggestion
        <version>13.0.2</version>
```

File: vaadin-archetype-application/src/main/resources/META-INF/archetype-post-generate.groovy [hunk 3]
```diff
@@ -0,0 +1,12 @@
+dir = new File(new File(request.outputDirectory), request.artifactId)
```
Comment (mstahv) [inline vaadin-archetype-application/src/main/resources/META-INF/archetype-post-generate.groovy]: If we plan to cut versions of archetypes with all platform releases, we can remove this file. If not, this part needs some more testing and should probably be disabled in certain environments where it fails.

We can probably do the testing with viritin archetypes if needed, but I think some Windows user reported a problem and from Eclipse at least the script isn't run at all.

File: vaadin-archetype-application/src/main/resources/archetype-resources/LICENSE.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
+This is free and unencumbered software released into the public domain.
```
Comment (mstahv) [inline vaadin-archetype-application/src/main/resources/archetype-resources/LICENSE.md]: I guess this was also copy pasted from viritin flow archetypes. I don't know what licenses archetypes usually put for the generated code if any. At least this should be fine as users can do what they want with this, even change the license.

File: vaadin-archetype-application/src/main/resources/archetype-resources/README.md [hunk 5]
```diff
@@ -1,55 +1,14 @@
...
 
-Template for a simple Vaadin application that only requires a Servlet 3.0 container to run.
+This project can be used as a starting point to create your own Vaadin application.
+It has the necessary dependencies and files to help you get started.
 
+To run the project, run `mvn jetty:run` and open [http://localhost:8080](http://localhost:8080) in browser.
 
-Workflow
-========
+To update to latest Vaadin release, issue `mvn versions:update-properties`
```
Comment (mstahv) [inline vaadin-archetype-application/src/main/resources/archetype-resources/README.md]:
```suggestion
To update to the latest available Vaadin release, issue `mvn versions:update-properties`
```

File: vaadin-archetype-application/src/main/resources/archetype-resources/pom.xml [hunk 6]
```diff
@@ -1,174 +1,114 @@
...
-	<packaging>war</packaging>
-	<version>\${version}</version>
-	<name>\${rootArtifactId}</name>
+    <properties>
+        <maven.compiler.source>1.8</maven.compiler.source>
+        <maven.compiler.target>1.8</maven.compiler.target>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <failOnMissingWebXml>false</failOnMissingWebXml>
+        
+        <vaadin.version>13.0.1</vaadin.version>
```
Comment (mstahv) [inline vaadin-archetype-application/src/main/resources/archetype-resources/pom.xml]:
```suggestion
        <vaadin.version>13.0.2</vaadin.version>
```

File: vaadin-archetype-application/src/main/resources/META-INF/archetype-post-generate.groovy (hunk 3 above)
Comment (pleku) [reply vaadin-archetype-application/src/main/resources/META-INF/archetype-post-generate.groovy]: Done

File: vaadin-archetype-application/src/main/resources/archetype-resources/LICENSE.md (hunk 4 above)
Comment (pleku) [reply vaadin-archetype-application/src/main/resources/archetype-resources/LICENSE.md]: Unlicensed should be used for any project templates like starters and archetypes.

File: vaadin-archetype-application/src/main/resources/META-INF/archetype-post-generate.groovy (hunk 3 above)
Comment (mstahv) [reply vaadin-archetype-application/src/main/resources/META-INF/archetype-post-generate.groovy]: I meant the whole archetype-post-generate.groovy file. Now the automatic upgrade to latest version probably don't work from CLI either.

File: pom.xml (hunk 1 above)
Comment (mstahv) [reply pom.xml]: BTW. This version should bi 13.0.2-SNAPSHOT in the SCM, until the actual archetypes are built.

File: vaadin-archetype-application/src/main/resources/archetype-resources/README.md [hunk 7]
```diff
@@ -1,55 +1,14 @@
...
-#set( $symbol_escape = '\' )
-${rootArtifactId}
-==============
+# Project Base for a Vaadin application
 
-Template for a simple Vaadin application that only requires a Servlet 3.0 container to run.
+This project can be used as a starting point to create your own Vaadin application.
+It has the necessary dependencies and files to help you get started.
 
+To run the project, run `mvn jetty:run` and open [http://localhost:8080](http://localhost:8080) in browser.
```
[the remark this answers is not in the document]
Comment (mstahv) [reply vaadin-archetype-application/src/main/resources/archetype-resources/README.md]: Did you do that on a generated project or directly from the archetype project? In other words, do you know how archetypes work?

[the remark this answers is not in the document]
Comment (ZheSun88) [reply vaadin-archetype-application/src/main/resources/archetype-resources/README.md]: No, I am not sure how it works.

[the remark this answers is not in the document]
Comment (mstahv) [reply vaadin-archetype-application/src/main/resources/archetype-resources/README.md]: https://maven.apache.org/guides/mini/guide-creating-archetypes.html

If you want to test it, install a snapshot locally first, by issuing mvn install in the top level (not in archetype resourses!). Then you can test it in a separate directory with "mvn archetype:generate" from the command line or via IDE.

I did a smoke test locally and it works fine on Mac at least. Even the upgrade to latest Vaadin version by the bit hacky groovy script.

[the remark this answers is not in the document]
Comment (ZheSun88) [reply vaadin-archetype-application/src/main/resources/archetype-resources/README.md]: i have also tried that in the top level (`/archetypes/*` )and got this error

```
[ERROR] Archetype IT 'basic' failed: java.io.IOException: Cannot run program "mvn" (in directory "C:\WorkFolder\archetypes\vaadin-archetype-application\target\test-classes\projects\basic\project\basic"): CreateProcess error=2, The system cannot find the file specified

```

File: vaadin-archetype-application/src/main/resources/archetype-resources/pom.xml (hunk 6 above)
Comment (pleku) [reply vaadin-archetype-application/src/main/resources/archetype-resources/pom.xml]: Done

File: vaadin-archetype-application/src/main/resources/archetype-resources/README.md (hunk 5 above)
Comment (pleku) [reply vaadin-archetype-application/src/main/resources/archetype-resources/README.md]: Done

File: pom.xml (hunk 1 above)
Comment (pleku) [reply pom.xml]: Done

File: vaadin-archetype-application/pom.xml (hunk 2 above)
Comment (pleku) [reply vaadin-archetype-application/pom.xml]: Done

File: vaadin-archetype-application/src/main/resources/archetype-resources/README.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (pleku) [reply vaadin-archetype-application/src/main/resources/archetype-resources/README.md]: This due to you having windows. 

I've removed the groovy file already.

[the remark this answers is not in the document]
Comment (mstahv) [reply vaadin-archetype-application/src/main/resources/archetype-resources/README.md]: And it happens as the archetype project as a basic smoke test built-in and it executed a hacky groove script that is now removed.

**Pull request merged**